frontend/src/metabase/query_builder/components/CompareAggregations/CompareAggregations.unit.spec.tsx

```diff
@@ -0,0 +1,208 @@
+      await userEvent.tab();
+
+      expect(input).toHaveValue(5);
+    });
+
+    it("does not allow non-integer values", async () => {
+      setup({ query: createQueryWithCountAggregation() });
+
+      const input = screen.getByLabelText("Previous period");
+
+      await userEvent.clear(input);
+      await userEvent.type(input, "1.234");
+      await userEvent.tab();
+
+      expect(input).toHaveValue(1);
+    });
+  });
+
+  describe("submit", () => {
+    it("is submittable by default", () => {
+      setup({ query: createQueryWithCountAggregation() });
+
+      expect(screen.getByLabelText("Previous period")).toHaveValue(1);
+      expect(screen.getByText("Previous value")).toBeInTheDocument();
+      expect(screen.getByText("Percentage difference")).toBeInTheDocument();
+      expect(screen.queryByText("Value difference")).not.toBeInTheDocument();
+      expect(screen.getByRole("button", { name: "Done" })).toBeEnabled();
+    });
+
+    it("disables the submit button when offset input is empty", async () => {
+      setup({ query: createQueryWithCountAggregation() });
+
+      const input = screen.getByLabelText("Previous period");
+
+      await userEvent.clear(input);
+
+      expect(screen.getByRole("button", { name: "Done" })).toBeDisabled();
+    });
+
+    it("disables the submit button when no columns are selected", async () => {
+      setup({ query: createQueryWithCountAggregation() });
+
+      await userEvent.click(screen.getByLabelText("Columns to create"));
+
+      const listBox = screen.getByRole("listbox");
+      await userEvent.click(within(listBox).getByText("Previous value"));
+      await userEvent.click(within(listBox).getByText("Percentage difference"));
+
+      expect(screen.getByRole("button", { name: "Done" })).toBeDisabled();
+    });
+
+    it("calls 'onSubmit' with single aggregation", async () => {
+      const { onSubmit } = setup({ query: createQueryWithCountAggregation() });
+
+      await userEvent.click(screen.getByLabelText("Columns to create"));
+
+      const listBox = screen.getByRole("listbox");
+      await userEvent.click(within(listBox).getByText("Previous value"));
+      await userEvent.click(screen.getByRole("button", { name: "Done" }));
+
+      const [aggregations] = onSubmit.mock.lastCall;
+      expect(onSubmit).toHaveBeenCalled();
+      expect(aggregations).toHaveLength(1);
+    });
+
+    it("calls 'onSubmit' with multiple aggregations", async () => {
+      const { onSubmit } = setup({ query: createQueryWithCountAggregation() });
+
+      await userEvent.click(screen.getByRole("button", { name: "Done" }));
+
+      const [aggregations] = onSubmit.mock.lastCall;
+      expect(onSubmit).toHaveBeenCalled();
+      expect(aggregations).toHaveLength(2);
+    });
+  });
+});
+
+function createQueryWithCountAggregation() {
+  return createQueryWithClauses({
+    aggregations: [{ operatorName: "count" }],
+  });
+}
+
+function createQueryWithCountAndSumAggregations() {
```

Comment:
These can be constants at the top of the file, no need for functions.
I'll refactor this.

Reply:
23b1b43365559f0bb0b4796b82da6b493c7f2052